server/src/main/java/org/elasticsearch/cluster/routing/OperationRouting.java

```diff
@@ -122,7 +124,13 @@ public GroupShardsIterator<ShardIterator> searchShards(
                 nodeCounts
             );
             if (iterator != null) {
-                var searchableShards = iterator.getShardRoutings().stream().filter(ShardRouting::isSearchable).toList();
+                var searchableShards = iterator.getShardRoutings().stream().filter((shardRouting -> {
```

Comment:
nit: might be a good chance to rename this since `searchable` has a specific meaning now.